fbs/matrix.fbs

```diff
@@ -22,39 +22,78 @@
 
 namespace NetEncoding;
 
-table Float32Array {
+table Float32FBArray {
   data: [float32];
 }
 
-table Uint32Array {
+table Uint32FBArray {
   data: [uint32];
 }
 
-table Int32Array {
+table Int32FBArray {
   data: [int32];
 }
 
-table Float64Array {
+table Float64FBArray {
   data: [float64];
 }
 
-table JSONEncodedArray {
+table JSONEncodedFBArray {
   // contains a UTF-8/JSON encoded array.  Used to store other
   // types (or polymorphic arrays)
   data: [uint8];
 }
 
-union TypedArray {
-  Float32Array,
-  Int32Array,
-  Uint32Array,
-  Float64Array,
-  JSONEncodedArray
+table CatInt8FBArray {  
+  data: [int8];
+  // "codes" is a UTF-8/JSON dictionary encoded as a byte array,
+  // Stores the mapping between codes and values
+  // JSON object schema is {[code: integer]: value: string}
```

Comment:
is `value` in the code table _always_ a string, or do you also use this type for other types (boolean, int, etc)?  Pandas Categoricals support all of the above which is why I ask.

If it is truly used _only_ for string categoricals, then you would likely benefit from encoding the code table as a flatbuffer array of struct, where the struct is { int, string}.  If it is actually polymorphic, then JSON makes sense.

In the past, we have seen categorical arrays that had int and bool category types.  But the overwhelming majority is string.  TL;DR - there may be more optimization available here if you want it.

Reply:
The intention was to allow polymorphic types that would get converted to strings in the fronted - typing the value as string here was an inaccuracy.